tabulate.py

```diff
@@ -593,6 +597,42 @@ def simple_separated_format(separator):
     )
 
 
+def _isnumber_with_thousands_separator(string):
+    """
+    >>> _isnumber_with_thousands_separator(".")
+    False
+    >>> _isnumber_with_thousands_separator("1")
+    True
+    >>> _isnumber_with_thousands_separator("1.")
+    True
+    >>> _isnumber_with_thousands_separator(".1")
+    True
+    >>> _isnumber_with_thousands_separator("1000")
+    False
+    >>> _isnumber_with_thousands_separator("1,000")
+    True
+    >>> _isnumber_with_thousands_separator("1,0000")
+    False
+    >>> _isnumber_with_thousands_separator("1,000.1234")
+    True
+    >>> _isnumber_with_thousands_separator(b"1,000.1234")
+    True
+    >>> _isnumber_with_thousands_separator("+1,000.1234")
+    True
+    >>> _isnumber_with_thousands_separator("-1,000.1234")
+    True
+    """
+    try:
+        string = string.decode()
+    except (UnicodeDecodeError, AttributeError):
+        pass
+
+    if re.match(_float_with_thousands_separators, string):
+        return True
+
+    return False
```

Comment:
This structure:

```py
if something:
    return True
else:
    return False
```

is a bit of a smell - `something` is probably bool-like so could be returned directly.

How about optimising and having one less `if` statement?

```suggestion
    return bool(re.match(_float_with_thousands_separators, string))
```